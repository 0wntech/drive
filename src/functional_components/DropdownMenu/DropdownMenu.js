import React, { Fragment } from 'react';
import PropTypes from 'prop-types';
import SvgDropdown from '../../assets/svgIcons/Dropdown';
import styles from './DropdownMenu.module.css';
import classNames from 'classnames';

function DropdownMenu({ options, renderOption, isExpanded, setExpanded }) {
    const defaultRender = (option) => {
        return (
            <div
                onClick={() => {
                    option.onClick();
                    setExpanded(!isExpanded);
                }}
                className={styles.option}
            >
                {option.label}
            </div>
        );
    };

    return (
        <Fragment>
            <SvgDropdown
                className={styles.icon}
                onClick={() => setExpanded(!isExpanded)}
            />
            <div
                className={classNames(styles.dropdownContent, {
                    [styles.hide]: !isExpanded,
                })}
            >
                {renderOption
                    ? options.map((option) => (
                          <div className={styles.option}>
                              {renderOption(option)}
                          </div>
                      ))
                    : options.map((option) => defaultRender(option))}
            </div>
        </Fragment>
    );
}

DropdownMenu.propTypes = {
    options: PropTypes.arrayOf(
        PropTypes.shape({
            label: PropTypes.string.isRequired,
            onClick: PropTypes.func.isRequired,
        })
    ),
    renderOption: PropTypes.func,
};

export default DropdownMenu;
