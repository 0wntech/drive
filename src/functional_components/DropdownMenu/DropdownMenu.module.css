.hide {
    display: none;
}

.container {
    position: relative;
    height: 100%;
}

.icon {
    cursor: pointer;
}

.dropdownContent {
    position: absolute;
    width: 100%;
    top: 100%;
    background-color: white;
    border-radius: 0px 0px 20px 20px;
    border-top: none;
    border-right: 0.5px solid #c4c4c4;
    border-bottom: 0.5px solid #c4c4c4;
    border-left: 0.5px solid #c4c4c4;
    box-sizing: border-box;
}

.option {
    padding: 5px 15px;
    font-family: 'Roboto';
    font-style: normal;
    font-weight: 300;
    font-size: 16px;
    line-height: 21px;
    letter-spacing: 0.05em;
    border-right: 0.5px solid #c4c4c4;
    border-left: 0.5px solid #c4c4c4;
    cursor: pointer;
}

.option:last-child {
    border-radius: 0px 0px 20px 20px;
    padding-bottom: 5px;
    border-bottom: 0.5px solid #c4c4c4;
}

.option:hover {
    background-color: #f2f2f2;
}
