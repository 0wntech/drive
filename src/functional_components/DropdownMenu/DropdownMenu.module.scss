@import '../../../styles/constants.scss';

.hide {
    display: none;
}

.head {
    height: initial !important;
    display: flex;
    align-items: center;
    justify-content: space-between;
    padding: 4px 8px;
    cursor: pointer;
}

.dropdownWrapper {
    display: flex;
    align-items: center;
    width: 100%;
    height: 100%;
    position: relative;
}

.iconWrapper {
    display: flex;
    justify-content: center;
    width: 30px;
}

.container {
    position: absolute;
    top: 6px;
    padding: 10px;
    width: calc(100% - 16px);
    justify-content: center;
    display: flex;
    flex-direction: column;
}

.active {
    padding: initial;
    background: #ffffff;
    border: solid #f4f4f4 10px;
    border-radius: 20px;
}

.icon {
    cursor: pointer;
}

.dropdownContent {
    position: relative;
    padding-bottom: 4px;
}

.option {
    padding: 5px 15px;
    font-family: 'Helvetica Neue Light';
    font-style: normal;
    font-weight: 300;
    font-size: 16px;
    line-height: 21px;
    letter-spacing: 0.05em;

    cursor: pointer;
}

.option:last-child {
    color: red;
    border-radius: 0px 0px 20px 20px;
    padding-bottom: 5px;
}

.option:hover {
    background-color: #DEEBFF;
}

@media (max-width: $screen_m) {
    .container {
        top: 0px;
        position: initial;
    }

    .iconWrapper {
        display: none;
    }

    .head {
        display: initial;
        margin-top: initial;
        padding: initial;
        justify-content: center;
        height: initial !important;
    }

    .dropdownWrapper {
        width: initial;
        position: initial;
    }
}
