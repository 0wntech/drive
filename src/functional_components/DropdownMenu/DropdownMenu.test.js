import React from 'react';
import DropdownMenu from './DropdownMenu';
import TestRenderer from 'react-test-renderer';

describe('DropdownMenu', () => {
    const DROPDOWN_OPTIONS = [
        { onClick: () => console.log('/profile'), label: 'Profile' },
        { onClick: () => console.log('test2'), label: 'Settings*' },
        { onClick: () => console.log('test2'), label: 'Notifications*' },
        { onClick: () => console.log('test2'), label: 'Contacts*' },
        { onClick: () => console.log('test2'), label: 'Logout' },
    ];

    it('should render correctly expanded', () => {
        const mockFunction = jest.fn();
        const menuHead = <p>Test</p>;
        const tree = TestRenderer.create(
            <DropdownMenu
                options={DROPDOWN_OPTIONS}
                isExpanded={true}
                setExpanded={mockFunction}
                menuHead={menuHead}
            />
        );
        expect(tree.toJSON()).toMatchSnapshot();
    });

    it('should render correctly not expanded', () => {
        const mockFunction = jest.fn();
        const tree = TestRenderer.create(
            <DropdownMenu
                options={DROPDOWN_OPTIONS}
                isExpanded={false}
                setExpanded={mockFunction}
            />
        );
        expect(tree.toJSON()).toMatchSnapshot();
    });
});
