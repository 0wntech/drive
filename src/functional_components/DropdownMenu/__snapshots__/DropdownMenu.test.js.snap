// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DropdownMenu should render correctly expanded 1`] = `
Array [
  <svg
    className="icon"
    fill="none"
    height={10}
    onClick={[Function]}
    width={14}
  >
    <path
      d="M0 0h14L7 10 0 0z"
      fill="#C4C4C4"
    />
  </svg>,
  <div
    className="dropdownContent"
  >
    <div
      className="option"
      onClick={[Function]}
    >
      Profile
    </div>
    <div
      className="option"
      onClick={[Function]}
    >
      Settings*
    </div>
    <div
      className="option"
      onClick={[Function]}
    >
      Notifications*
    </div>
    <div
      className="option"
      onClick={[Function]}
    >
      Contacts*
    </div>
    <div
      className="option"
      onClick={[Function]}
    >
      Logout
    </div>
  </div>,
]
`;

exports[`DropdownMenu should render correctly not expanded 1`] = `
Array [
  <svg
    className="icon"
    fill="none"
    height={10}
    onClick={[Function]}
    width={14}
  >
    <path
      d="M0 0h14L7 10 0 0z"
      fill="#C4C4C4"
    />
  </svg>,
  <div
    className="dropdownContent hide"
  >
    <div
      className="option"
      onClick={[Function]}
    >
      Profile
    </div>
    <div
      className="option"
      onClick={[Function]}
    >
      Settings*
    </div>
    <div
      className="option"
      onClick={[Function]}
    >
      Notifications*
    </div>
    <div
      className="option"
      onClick={[Function]}
    >
      Contacts*
    </div>
    <div
      className="option"
      onClick={[Function]}
    >
      Logout
    </div>
  </div>,
]
`;
